chainerx_cc/examples/mnist_cc/mnist.cc

```diff
@@ -0,0 +1,79 @@
+#include "mnist.h"
+
+#include <cassert>
+#include <cstdint>
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include "chainerx/array.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/shape.h"
+
+namespace chx = chainerx;
+
+namespace {
+
+chx::Array ReadArray(std::ifstream& ifs, const chx::Shape& shape) {
+    int64_t n = shape.GetTotalSize();
+
+    std::shared_ptr<uint8_t> data{new uint8_t[n], std::default_delete<uint8_t[]>{}};
+    ifs.read(reinterpret_cast<char*>(data.get()), n);
+
+    return chx::FromContiguousHostData(shape, chx::Dtype::kUInt8, static_cast<std::shared_ptr<void>>(data), chx::GetDefaultDevice());
+}
+
+int32_t ReadHeader(std::ifstream& ifs) {
```

Comment:
How about `ReadInt32`? This function name is misleading because a single int32_t is not the entire header.